src/QMCDrivers/SimpleFixedNodeBranch.h

```diff
@@ -116,30 +123,41 @@ struct SimpleFixedNodeBranch : public QMCTraits
   typedef TinyVector<int, B_IPARAM_MAX> IParamType;
   IParamType iParam;
 
-  /*! enum for vParam */
-  enum
+  /** enum for vParam 
+   *
+   *  Easy serialization is a relatively minor concern compared to the 
+   *  annoyance this causes elsewhere.
+   */
+  enum class SBVP
   {
-    B_TAU = 0,
-    B_TAUEFF,
-    B_ETRIAL,
-    B_EREF,
-    B_ENOW,
-    B_BRANCHMAX,
-    B_BRANCHCUTOFF,
-    B_BRANCHFILTER,
-    B_SIGMA2,
-    B_ACC_ENERGY,
-    B_ACC_SAMPLES,
-    B_FEEDBACK,
-    B_FILTERSCALE,
-    B_VPARAM_MAX = 17
+    TAU = 0,
+    TAUEFF,
+    ETRIAL,
+    EREF,
+    ENOW,
+    BRANCHMAX,
+    BRANCHCUTOFF,
+    BRANCHFILTER,
+    SIGMA2,
+    ACC_ENERGY,
+    ACC_SAMPLES,
+    FEEDBACK,
+    FILTERSCALE,
+    VPARAM_MAX = 17 // four extra, why? Sloppy or undocumented hack?
   };
 
   /** controlling parameters of real type
    *
    * Mostly internal
    */
-  typedef TinyVector<FullPrecRealType, B_VPARAM_MAX> VParamType;
+  struct VParamType : public std::array<FullPrecRealType, static_cast<size_t>(SBVP::VPARAM_MAX)>
```

Comment:
```
using VParamType = std::array<FullPrecRealType, static_cast<size_t>(SBVP::VPARAM_MAX)>
```
Is this better?

Reply:
I finished what I was refactoring toward. Removed the `operator[](int)` vParamType now has a type safe operator[] i.e. you must use a SBVP value. The vParam enum doesn't pollute the qmcplusplus namespace.

The other advantage is any C++ programmer knows how std::array works. Not so TinyVector.

Long term motive, I can now get the enum type the vParam uses so a pretty printer for the debugger can be written and I don't have to leave the declaration of the enum open.

Reply:
TinyVector should be only used for tight math. All the generic use of array with known size at compile time should adopt std::array. Your way of wrapping std::array as VParamType still leak public accessors like `at()`, but we at least got some protection.